beta-pipelines/skylab/slidetags/scripts/spatial-count.wdl

```diff
@@ -0,0 +1,95 @@
+version 1.0
+
+task count {
+  input {
+    Array[String] fastq_paths
+    Array[String] pucks
+    Int mem_GiB
+    Int disk_GiB
+    String docker
+  }
+  command <<<
+    set -euo pipefail
+    set -x
+
+    echo "<< starting spatial-count >>"
+
+    gcloud config set storage/process_count 16
+    gcloud config set storage/thread_count  2
+
+    # Download the script -- put this script into a docker
+    wget https://raw.githubusercontent.com/MacoskoLab/Macosko-Pipelines/5c74e9e6148102081827625b9ce91ec2b7ba3541/spatial-count/spatial-count.jl
+    ls 
+
+    echo "FASTQs: ~{length(fastq_paths)} paths provided"
+    echo "Pucks: ~{length(pucks)} puck(s) provided"
+
+    # Assert that the fastqs exist
+    fastqs=(~{sep=' ' fastq_paths})
+    for fastq in "${fastqs[@]}" ; do
+        if ! gsutil stat "$fastq" &> /dev/null ; then
+            echo "ERROR: gsutil stat command failed on fastq $fastq"
+            exit 1
+        fi
+    done
+
+    # Download the fastqs
+    echo "Downloading fastqs:"
+    mkdir fastqs
+    gcloud storage cp ~{sep=' ' fastq_paths} fastqs
+
+    # Assert that the pucks exist
+    pucks=(~{sep=' ' pucks})
+    for puck in "${pucks[@]}" ; do
+        if ! gsutil stat "$puck" &> /dev/null ; then
+            echo "ERROR: gsutil stat command failed on puck $puck"
+            exit 1
+        fi
+    done
+
+    # Download the pucks
+    echo "Downloading pucks:"
+    mkdir pucks
+    gcloud storage cp ~{sep=' ' pucks} pucks
+
+    # Run the script
+    echo ; echo "Running spatial-count.jl"
+    ## julia --threads=4 /spatial-count.jl fastqs pucks .
+    julia --threads=4 spatial-count.jl fastqs pucks .
+    echo "TEST"
+    ls
+    
+    if [[ -f SBcounts.h5 ]] ; then
```

Comment:
Are there issues with things not completing as expected and not failing with a meaningful error message? Just curious

Reply:
The script is still under development so I think that may be the case sometimes